src/dispatcher.coffee

```diff
@@ -1,44 +1,113 @@
-angular.module('dispatcher', []).service 'dispatcher', ->
-    _callbacks = []
-    _isDispatching = false
-
-    ###*
-    Register a store callback and returns a deregistration method.
-    ###
-    register: (callback) ->
-        _callbacks.push(callback)
-        return =>
-            @unregister(callback)
-
-    ###*
-    Unregister a store.
-    ###
-    unregister: (callback) ->
-        index = _callbacks.indexOf(callback)
-
-        if index < 0
-            throw new Error('dispatcher.unregister(...): callback argument is not a registered callback.')
-
-        _callbacks.splice(index, 1)
-
-    ###*
-    Start dispatching action and payload to the stores
-    ###
-    dispatch: (action, payload) ->
-        if _isDispatching
-            throw new Error('dispatcher.dispacth(...): Cannot dispacth while dispatching')
-
-        _isDispatching = true
-
-        try
-            for callback in _callbacks
-                callback(action, payload)
-
-        finally
-            _isDispatching = false
-
-    ###*
-    Return true if dispatcher is in the middle of a dispatch
-    ###
-    isDispatching: ->
-        return _isDispatching
\ No newline at end of file
+angular.module('dispatcher', []).service 'dispatcher',
+    (
+        dataContract
+    ) ->
+        @_callbacks = {}
+        @_isHandled = {}
+        @_isPending = {}
+        @_lastID = number
```

Comment:
```
@_lastID = 0
```

i don't think `number` is defined, which would make line [38](https://github.com/debrice/angularjs-dispatcher/pull/1/files#diff-43f3cc78d366e353207a8119c6cfc0aaR38) result in `ID_NaN`